tensorflow_addons/metrics/cohens_kappa.py

```diff
@@ -0,0 +1,140 @@
+
+
+@keras_utils.register_keras_custom_object
+class CohensKappa(Metric):
+  """Computes Kappa score between two raters.
+
+  The score lies in the range [-1,1]. A score of -1 represents
+  complete disagreement between two raters whereas a score of 1 
+  represents complete agreement between the two raters. 
+  A score of 0 means agreement by chance.
+  
+  Note: As of now, this implementation considers all labels 
+  while calculating the Cohen's Kappa score.
+
+  Usage:
+  ```python
+  actuals = np.array([4, 4, 3, 4, 2, 4, 1, 1], dtype=np.int32)
+  preds = np.array([4, 4, 3, 4, 4, 2, 1, 1], dtype=np.int32)
+
+  m = tf.keras.metrics.CohensKappa()
+  m.update_state(actuals, preds, "quadratic")
+  print('Final result: ', m.result().numpy()) # Result: 0.68932
+  ```
+  Usage with tf.keras API:
+  ```python
+  model = keras.models.Model(inputs, outputs)
+  model.add_metric(tf.keras.metrics.CohensKappa(name='kp_score')(outputs))
+  model.compile('sgd', loss='mse')
+  ```
+
+  Args:
+    y1 : array, shape = [n_samples]
+      Labels assigned by the first annotator.
+    y2 : array, shape = [n_samples]
+      Labels assigned by the second annotator. The kappa statistic is
+      symmetric, so swapping ``y1`` and ``y2`` doesn't change the value.
+    sample_weight(optional) : None or str 
+      A string denoting the type of weighting to be used.
+      Valid values for this parameter are [None, 'linear', 'quadratic'].
+      Default value is None.
+
+
+  Returns:
+    kappa_score : float
+      The kappa statistic, which is a number between -1 and 1. The maximum
+      value means complete agreement; zero or lower means chance agreement.
+
+
+  Raises:
+    ValueError: If the value passed for `sample_weight` is invalid
+      i.e. not any one of [None, 'linear', 'quadratic']
+
+  """
+  def __init__(self, name='cohens_kappa', dtype=tf.float32,):
+    super(CohensKappa, self).__init__(name=name, dtype=dtype)
+    self.kappa_score = self.add_weight('kappa_score', 
+                                       initializer=None)
+
+  def update_state(self, y_true, y_pred, sample_weight=None):
+    y_true = tf.cast(y_true, dtype=tf.int32)
+    y_pred = tf.cast(y_pred, dtype=tf.int32)
+
+    # 1. Get the confusion matrix
+    conf_mtx = confusion_matrix(labels=y_true, predictions=y_pred)     
+    nb_ratings = tf.shape(conf_mtx)[0]
+    
+    # 2. Create a weight matrix
+    if sample_weight is None:
+      weight_mtx = tf.ones([nb_ratings, nb_ratings], dtype=tf.int32)
+      diagonal = tf.zeros([5], dtype=tf.int32)
+      weight_mtx = tf.linalg.set_diag(weight_mtx, diagonal=diagonal)
+      weight_mtx = tf.cast(weight_mtx, dtype=tf.float32)
+    
+    elif sample_weight == "linear": 
+      weight_mtx = tf.ones([nb_ratings, nb_ratings], dtype=tf.int32)
+      weight_mtx += tf.range(nb_ratings, dtype=tf.int32)
+      weight_mtx = tf.cast(weight_mtx, dtype=tf.float32)
+      weight_mtx = tf.abs(weight_mtx - K.transpose(weight_mtx))
```

Comment:
This is the only line that changes based on `sample_weight`. It would be better to refactor the code to combine the same code, and handle only `abs`/`pow` in a condition.